enhancements/network/egress-router.md

```diff
@@ -0,0 +1,200 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Egress traffic is traffic going from OpenShift pods to external systems,
+outside of OpenShift. There are a few  options for enabling egress traffic,
+such as allow access to external systems from OpenShift physical node IPs, use
+EgressFirewall, EgressIPs or in our case, EgressRouter.
+
+In enterprise environments, egress routers are often preferred. They
+allow granular access from a specific pod, group of pods, or project to an
+external system or service. Access via node IP means all pods running on a
+given node can access external systems.
+
+An egress router is a pod that has two interfaces (eth0) and (macvlan0). Eth0
+is on the cluster network in OpenShift (internal) and macvlan0 has an IP and
+gateway from the external physical network.
+
+Pods can access the egress router service thus enabling them to access
+external services. The egress router acts as a bridge between pods and an
+external system.
+
+Traffic going out the egress router goes via node, but it will have the MAC
+address of the macvlan0 interface inside the egress router.
+
+In openshift-sdn, the egress router was implemented by adding an annotation to
+allow a pod to request a macvlan interface. In order to avoid repeating this
+behavior in ovn-kubernetes, we'd be requesting such interface using multus to
+ensure feature-parity with openshift-sdn.
+
+## Motivation
+
+* The egress router images allow people to run containers that have defined
+  external addresses.
+
+  Using network policies you can control the scope of who can talk to egress
+  routers and thus, when they send traffic outside the cluster it's possible
+  to identify the containers in the cluster which are sending it. This allows
+  to use external firewalls for auditing and filtering.
+
+  Using multus for requesting the macvlan interface also opens up the
+  possibility of use different interface types in the future.
+
+* Ensure that we have images that offer equivalent capabilities to the existing
+  egress router images when running ovn-kubernetes.
+
+### Goals
+
+- Create an egress router CNI plugin
+- Add feature parity with openshift-sdn's egress router.
+
+## Proposal
+
+## Design Details
+
+We'll be implenting this feature using an additional specific CNI plugin for
+it, whose [github repository](https://github.com/openshift/egress-router-cni/)
+is already created.
+
+Both kubernetes' pod and multus' net-attach-def are namespaced objects. Due to a
+security limitation in OCP, Pod yaml and net-attach-def object should be in the
+same namespace, thus egress-router-cni would net to take this into
+consideration and put them in the same namespace as well.
+
+The egress router pod would have multus annotations saying to use the
+egress-router-cni plugin, and then it would create a macvlan interface
+(or, in the future, maybe ipvlan or Amazon ENI).
```

Comment:
The "it" there sounds like you're saying the egress router pod is creating the macvlan interface, but it's the CNI plugin that does that.

Reply:
I'll rephrase the paragraph, thanks!